### ISSUE TITLE
Unhandled exception when `LockCachingAudioSource` receives an HTTP response without `Content-Length` header

### ISSUE THREAD

**Issue by agersant** [ryanheise/just_audio#570]
**Which API doesn't behave as documented, and how does it misbehave?**
`AudioPlayer.play()` can result in an unhandled exception when attempting to play a `LockCachingAudioSource` - if the remote host serving audio does not respond with a `content-length` header.

**Minimal reproduction project**
https://github.com/agersant/just_audio/tree/just-audio-issue-570

**To Reproduce (i.e. user steps, not code)**
1. Run the application from the `just_audio_background/example` directory
2. Click the `!` icon in the middle of the screen
3. Observe crash

**Error messages**

Error message: `Bad end position: -1`
Call stack(s):
```
#0      _HttpOutgoing.addStream.<anonymous closure> (dart:_http/http_impl.dart:1810:9)
#1      _rootRunBinary (dart:async/zone.dart:1452:47)
#2      _CustomZone.runBinary (dart:async/zone.dart:1342:19)
#3      _FutureListener.handleError (dart:async/future_impl.dart:175:22)
#4      Future._propagateToListeners.handleError (dart:async/future_impl.dart:779:47)
#5      Future._propagateToListeners (dart:async/future_impl.dart:800:13)
#6      Future._completeError (dart:async/future_impl.dart:610:5)
#7      Future._asyncCompleteError.<anonymous closure> (dart:async/future_impl.dart:666:7)
#8      _rootRun (dart:async/zone.dart:1428:13)
#9      _CustomZone.run (dart:async/zone.dart:1328:19)
#10     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
#11     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)
#12     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#13     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
```

```
_proxyHandlerForSource.handler (c:\Users\agersant\Documents\GitHub\just_audio\just_audio\lib\just_audio.dart:2942)
<asynchronous gap> (Unknown Source:0)
```

**Expected behavior**
There are no errors and the audio source can be heard.

**Screenshots**
N/A

**Desktop (please complete the following information):**
- OS: Windows 10
- Browser: N/A

**Smartphone (please complete the following information):**
- Device: Pixel 3a emulator
- Android 11

**Flutter SDK version**
```
[√] Flutter (Channel stable, 2.0.5, on Microsoft Windows [Version 10.0.19042.1288], locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.61.2)
[√] Connected device (3 available)
```

**Additional context**
Originally mentioned as a follow-up to https://github.com/ryanheise/just_audio/issues/569

**Comment by agersant** [comment]
Thank you for the quick fix! 

✔ I tested it on my original use-case (the Polaris server) and it worked great: played audio and did cache it as intended. The `originSupportsRangeRequests` check correctly picks up that the server supports range requests. As a side note, checking for the `accept-range` header might miss some servers that do support range requests. (see [HTTP Spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) 14.5). (although it's not an issue for my own use case)

✔ I also tested the `https://filesamples.com/samples/audio/mp3/sample4.mp3` example and verified that audio played with no errors.

**Comment by ryanheise** [comment]
Glad to know it's working so far :+1:

> As a side note, checking for the `accept-range` header might miss some servers that do support range requests. (see [HTTP Spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) 14.5).

Thanks for pointing this out. Hopefully your comment will help either me or someone else to create a pull request in the future once someone runs into the issue.

**Comment by ryanheise** [comment]
This fix is now released in 0.9.17. I'll close this issue as resolved even though there is another edge case I haven't dealt with, but it will be easier to deal with it once someone runs into the issue and creates a new issue with the details.

**Comment by agersant** [comment]
Thank you!
